src/blockrelay/graphene.cpp

```diff
@@ -496,33 +511,56 @@ bool CGrapheneBlock::process(CNode *pfrom,
         }
 
         // Add full transactions included in the block
+        CTransactionRef coinbase = nullptr;
         if (!collision)
         {
             for (auto &tx : vAdditionalTxs)
             {
                 const uint256 &hash = tx->GetHash();
                 uint64_t cheapHash = hash.GetCheapHash();
 
+                if (tx->IsCoinBase())
+                    coinbase = tx;
+
                 auto ir = mapPartialTxHash.insert(std::make_pair(cheapHash, hash));
                 if (!ir.second)
                     collision = true;
             }
         }
 
+        if (coinbase == NULL)
+            throw std::runtime_error("No coinbase transaction found in graphene block");
+
```

Comment:
I think the coinbase check could be moved up to the same place where we do it in thinblocks... Take a look at IsThinBlockValid() in thinblock.cpp.

Reply:
I suppose it depends on what your intent is though...are you expecting the graphene block to always have the coinbase?  In thinblocks we do, in compact blocks they don't force that but it eventually gets caught later because the block won't reconstruct...in graphene now, i can't remember if you MUST send the coinbase in the grapheneblock?

Reply:
For the purpose of CTOR it is essential that there exist a coinbase at this point and that I place it in the first position in the transaction vector. Otherwise, I cannot be certain that the ordering is correct in time for the Merkle root check.

Reply:
One other note though, throwing exceptions can be an easy attack vector for filling up log files.  It would be better to have  a LOG(GRAPHENE, "your message...") statement with a false return...so it only gets logged if weve got logging on.

Reply:
Sure, I can make that change.